networkit/cpp/centrality/DynApproxBetweenness.cpp

```diff
@@ -1,178 +1,268 @@
+                                           bool storePredecessors, double universalConstant)
+    : Centrality(G, true), epsilon(epsilon), delta(delta), storePreds(storePredecessors),
+      universalConstant(universalConstant) {}
+
+count DynApproxBetweenness::computeVDdirected() {
+    StronglyConnectedComponents scc(G);
+    scc.run();
+    count sccNum = scc.numberOfComponents();
+    Graph sccDAG(sccNum, false, true);
+    G.forEdges([&](node u, node v) {
+        if (scc.componentOfNode(u) != scc.componentOfNode(v)) {
+            if (!sccDAG.hasEdge(scc.componentOfNode(u) - 1, scc.componentOfNode(v) - 1)) {
+                sccDAG.addEdge(scc.componentOfNode(u) - 1, scc.componentOfNode(v) - 1);
+            }
+        }
+    });
 
+    TopologicalSort topSort = TopologicalSort(sccDAG);
+    topSort.run();
+    std::vector<node> sorted = topSort.getResult();
+
+    std::vector<bool> marked1(G.upperNodeIdBound()), marked2(G.upperNodeIdBound());
+    std::vector<node> sources(sccNum, 0);
+    G.forNodes([&](node u) {
+        count x = scc.componentOfNode(u) - 1;
+        if (sources[x] == 0) {
+            sources[x] = u;
+        }
+    });
+    // we compute the bound on VD for each SCC
+    std::vector<count> vdapproxs(sccNum);
+    std::queue<node> q, qNext;
+    for (count i = 0; i < sccNum; i++) {
+        count source = sources[sorted[i]];
+        // forward BFS
+        marked1[source] = true;
+        count dist1 = 0;
+        q.push(source);
+        do {
+            node u = q.front();
+            q.pop();
+            G.forNeighborsOf(u, [&](node v) {
+                if (!marked1[v] && scc.componentOfNode(u) == scc.componentOfNode(v)) {
+                    qNext.push(v);
+                    marked1[v] = true;
+                }
+            });
+            if (q.empty() && !qNext.empty()) {
+                q.swap(qNext);
+                ++dist1;
+            }
+        } while (!q.empty());
+
+        // backward BFS
+        marked2[source] = true;
+        count dist2 = 0;
+        q.push(source);
+        do {
+            node u = q.front();
+            q.pop();
+            G.forInNeighborsOf(u, [&](node v) {
+                if (!marked2[v] && scc.componentOfNode(u) == scc.componentOfNode(v)) {
+                    qNext.push(v);
+                    marked2[v] = true;
+                }
+            });
+            if (q.empty() && !qNext.empty()) {
+                q.swap(qNext);
+                ++dist2;
+            }
+        } while (!q.empty());
+        // source node is not taken into account in any of the two approximations
+        vdapproxs[sorted[i]] = dist1 + dist2 + 1;
+        if (vdapproxs[sorted[i]] == 0) // for singletons
+            vdapproxs[sorted[i]] = 1;
+    }
+    // starting from the bottom, we compute all the cumulative vd approximations
+    count vd = vdapproxs[0];
+    std::vector<count> vdSuccessors(sccNum, 0);
+    for (count i = 0; i <= sccNum - 1; i++) {
```

Comment:
This is more readable, I think:
```suggestion
    for (count i = 0; i < sccNum; i++) {
```